### PR TITLE
Bring PWA feature management in-house

### PR DESCRIPTION
**Pull request by claudiulodro**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR removes Progressive WP as the plugin we use for managing PWA features and brings that functionality into the Newspack plugin (see PWA class).

As part of this work I've simplified the wizard down to one screen, as the only thing that needs to be managed is a site icon:

Screen Shot 2019-08-31 at 8 17 39 AM

### How to test the changes in this Pull Request:

1. You'll need https for this to work properly. I tested with jurassic.ninja, but if you have https on your local environment that would work also.
2. Navigate to the Performance screen in the Newspack dashboard. You should observe a screen like the one pictured above, with a useful status message and ability to upload a site icon.
3. In chrome://flags, [enable these settings](https://user-images.githubusercontent.com/7317227/63796123-46465f00-c8ba-11e9-8d32-7a5b33439313.png).
4. On the frontend, you should see a (+) and have ability to install the site:
Screen Shot 2019-08-31 at 8 18 11 AM

5. In dev tools, go to Applications and toggle offline on:
Screen Shot 2019-08-31 at 8 17 08 AM

6. Visit a page you've already visited. It should be available. Visit a page you haven't visited; it should give PWA's default offline message for now (until our theme has an `offline.php` template).

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by claudiulodro** [comment]
Would it look nicer if site icon was a separate tab?

**Comment by claudiulodro** [comment]
> When I visit a page offline that hasn't been visited before, I get the "Cart" page. Is this expected? I assume when the theme adds an offline template this becomes a non-issue, but is there anything to how the offline page is chosen if no template exists?

Sorry, I probably should have specified: step 0 is deactivate Progressive WP. I believe you had that active, as using a page for the offline screen is one of its features. With just this PR, you should be seeing a gray-ish default-wp-admin-looking offline screen, and not an existing page.

**Comment by claudiulodro** [comment]
> What's the story with the Chrome flags, and how does this relate to the experience of a normal site visitor who wouldn't be expected to interact with these manually?

On desktop, the ability to install sites on home screen is turned off by default (on mobile it isn't). This may change in the future, but for now installing a site is really just a mobile thing. 

The first flag enables the ability to install sites. The second flag is really just for our debugging use. Normally when a user rejects installing a site, Chrome won't ask again for a while; enabling the flag makes it so that Chrome always asks.

**Comment by claudiulodro** [comment]
> The offline toggle in dev tools simulates an offline state, as opposed to activating the feature, correct? To ask another way, is going into Airplane Mode on a phone identical to checking that box?

Airplane mode should work fine for testing it also. The main reason I recommended using the Chrome setting instead of e.g. deactivating your wifi is because on a local environment deactivating your wifi will still load any page since it's not served from the internet. :)

**Comment by claudiulodro** [comment]
> To confirm, Automattic/newspack-theme#355 is an optional addition, not a requirement for testing this branch?

Correct. It lets PWA cache the AMP scripts for offline use. There may be issues around e.g. slideshows not working if the AMP scripts are not cached, but in general things should still work.

**Comment by claudiulodro** [comment]
> Besides the addition of the offline template, is there any additional theme work that needs to happen to complete PWA support?

The only thing remaining is the offline template. The PWA plugin should automatically pick up and use the correct template if it's named `offline.php`.

File: assets/wizards/performance/style.scss [hunk 1]
```diff
@@ -21,3 +21,19 @@
 		color: $muriel-gray-500;
 	}
 }
+
+.newspack-performance-wizard__status {
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply assets/wizards/performance/style.scss]: I agree, a notice feature would be super handy in the wizards and used in a lot of places.

Screen Shot 2019-09-03 at 9 20 31 AM

Do you think we should use the notices from this PR for e.g. the pictured notice? Alternately, do you think we should use the notice style seen in the screenshot everywhere (including this PR)? Standardizing everything to remove all the special notice styles from the wizards is a good idea.

**Comment by jeffersonrabb** [comment]
> When I visit a page offline that hasn't been visited before, I get the "Cart" page. Is this expected? I assume when the theme adds an offline template this becomes a non-issue, but is there anything to how the offline page is chosen if no template exists?

This seems to be resolved now, after turning PWA back on (not sure how that happened) and adding `use \WP_Service_Worker_Caching_Routes;` to the top of `class-pwa`. Now seeing the correct default Offline notifications.

**Comment by jeffersonrabb** [comment]
> Airplane mode should work fine for testing it also. The main reason I recommended using the Chrome setting instead of e.g. deactivating your wifi is because on a local environment deactivating your wifi will still load any page since it's not served from the internet. :)

I've been testing this PR primarily on a live site anyway.

File: assets/wizards/performance/views/intro/index.js [hunk 2]
```diff
@@ -56,15 +47,34 @@ class Intro extends Component {
...
+							<strong>
+								{ __( 'PWA options have been automatically set up for you.' ) }
+							</strong>
+						</p>
 					</div>
-					<p className="newspack-plugin-description">
-						{ __(
-							'Basic PWA options have been automatically set up for you. Advanced options are available in the Progressive WP dashboard.'
+					<div className='newspack-performance-wizard__status'>
+						<h3>{ __( 'Status' ) }</h3>
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply assets/wizards/performance/views/intro/index.js]: https://github.com/Automattic/newspack-plugin/pull/253/commits/7912a4ae0f2e96049c1e1ec186c2456c0d96b9a5#diff-8f62487fc5909334e1fe3341f766dbefL57

File: assets/wizards/performance/style.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (claudiulodro) [reply assets/wizards/performance/style.scss]: Made them available in https://github.com/Automattic/newspack-plugin/pull/253/commits/7912a4ae0f2e96049c1e1ec186c2456c0d96b9a5 and made https://github.com/Automattic/newspack-plugin/issues/261 as a follow-up task.

File: assets/wizards/performance/index.js [hunk 3]
```diff
@@ -103,65 +103,12 @@ class PerformanceWizard extends Component {
...
-								buttonAction={ () =>
-									this.updateSettings(
-										'push_notifications',
-										'push_notification_server_key',
-										'push_notification_sender_id'
-									).then( () => ( window.location = newspack_urls[ 'dashboard' ] ) )
-								}
 								settings={ settings }
 								updateSetting={ this.updateSetting }
+								buttonText={ __( 'Save' ) }
```
[the remark this answers is not in the document]
Comment (claudiulodro) [reply assets/wizards/performance/index.js]: I've made this update in https://github.com/Automattic/newspack-plugin/pull/253/commits/45cbc7d99d6511c91a65c67d603f8b6ccd0942a0. The site icon will save automatically every time a change is made to it.

**Comment by claudiulodro** [comment]
Good catch. I've removed the console log in https://github.com/Automattic/newspack-plugin/pull/253/commits/7f90e5de5dedb19a7effdff421151796baa80b2e.

Regarding the square icon, it is a hard requirement that the icon be square (do you have any apps with non-square icons on your phone?? :P). I definitely think we should improve those error banners, and I've created an issue around it at https://github.com/Automattic/newspack-plugin/issues/263

**Pull request merged**